src/pkg/k8s.go

```diff
@@ -367,8 +369,8 @@ func (s *JobRunner) CreatePod(config *corev1.Pod) (*corev1.Pod, error) {
 
 func (s *JobRunner) WaitForPod(podConfig *corev1.Pod, timeout time.Duration) error {
 	s.logger.Debug().Msgf("Waiting for pod %s/%s to be ready in %s ...", podConfig.Namespace, podConfig.Name, timeout)
-	return wait.PollImmediate(time.Second, timeout, func() (bool, error) {
-		pod, err := s.clientset.CoreV1().Pods(podConfig.Namespace).Get(context.TODO(), podConfig.Name, metav1.GetOptions{})
+	return wait.PollUntilContextTimeout(context.TODO(), time.Second, timeout, true, func(ctx context.Context) (bool, error) {
```

Comment:
What the purpose of this change?

Reply:
@rocktavious ~~`PollImmediate` is deprecated in k8s.io/v0.26.11 and the docs recommend replacing it with `PollUntilContextTimeout`~~ This is not true, it's deprecated in `v0.29.1` which is what GoLand was showing me. 

I am going to revert the change.